server/src/main/java/org/opensearch/action/admin/indices/view/SearchViewAction.java

```diff
@@ -0,0 +1,138 @@
+    public static final String NAME = "cluster:admin:views:search";
+
+    private SearchViewAction() {
+        super(NAME, SearchResponse::new);
+    }
+
+    /** Given a search request, creates a ViewSearchRequest */
+    public static Request createRequestWith(final String view, final SearchRequest searchRequest) throws IOException {
+        final BytesStreamOutput savedSearchRequest = new BytesStreamOutput();
+        searchRequest.writeTo(savedSearchRequest);
+        savedSearchRequest.writeString(view);
+
+        final BytesStreamInput input = new BytesStreamInput(savedSearchRequest.bytes().toBytesRef().bytes);
+        return new Request(input);
+    }
+
+    /**
+     * Wraps the functionality of search requests and tailors for what is available
+     * when searching through views
+     */
+    @ExperimentalApi
+    public static class Request extends SearchRequest {
+
+        private final String view;
+
+        public Request(final String view) {
+            super();
+            this.view = view;
+        }
+
+        public Request(final StreamInput in) throws IOException {
+            super(in);
+            view = in.readString();
+        }
+
+        public String getView() {
+            return view;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request that = (Request) o;
+            return view.equals(that.view) && super.equals(that);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(view, super.hashCode());
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            final Function<String, String> unsupported = (String x) -> x + " is not supported when searching views";
+            ActionRequestValidationException validationException = super.validate();
+
+            if (scroll() != null) {
+                validationException = addValidationError(unsupported.apply("Scroll"), validationException);
+            }
+
+            // TODO: Filter out anything additional search features that are not supported
+
+            if (Strings.isNullOrEmpty(view)) {
+                validationException = addValidationError("View is required", validationException);
+            }
+
+            return validationException;
+        }
+
+        @Override
+        public void writeTo(final StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(view);
+        }
+
+        @Override
+        public String toString() {
+            return super.toString().replace("SearchRequest{", "SearchViewAction.Request{view=" + view + ",");
+        }
+    }
+
+    /**
+     * Transport Action for searching a View
+     */
+    public static class TransportAction extends HandledTransportAction<Request, SearchResponse> {
```

Comment:
Should this go under SEARCH or INDEX_SEARCHER threadpool?

Reply:
TransportClusterManagerNodeAction asks for a threadpool  to execute actions on, no additional threads are spun up or used for SearchViewActions

Reply:
aah.. didn;t know that. Thank you for explanation!